### PR TITLE
feat(issue-details): Add a foldable data section component

### PR DESCRIPTION
**Pull request by leeandher**
Super simple folding section. Didn't make progress on removing the margins for the independent data sections just yet, but at least checking this in will allow its use in the next week while I'm out

https://github.com/user-attachments/assets/2594daa9-e9fe-40ee-b83b-fd70313b01f9

### CODE REVIEW

File: static/app/views/issueDetails/streamline/foldSection.tsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+}: FoldSectionProps) {
+  const [isCollapsed, setIsCollapsed] = useLocalStorageState(
+    `${LOCAL_STORAGE_PREFIX}${sectionKey}`,
+    initialCollapse
+  );
+
+  const toggleCollapse = useCallback(
+    (e: React.MouseEvent) => {
+      e.preventDefault();
+      setIsCollapsed(!isCollapsed);
```
Comment (malwilley) [inline static/app/views/issueDetails/streamline/foldSection.tsx:41]: Should use the callback when the state change is dependent on the old state: `setIsCollapsed(collapsed => !collapsed)`

File: static/app/components/globalDrawer/index.stories.tsx
Comment (malwilley) [inline static/app/components/globalDrawer/index.stories.tsx:1]: These changes look unrelated?

**Review by malwilley — COMMENTED** [review]
This might be planned future behavior, but I'd expect to see some hover/pressed states? You might be able to use `InteractionStateLayer` if you want the whole header to appear clickable.

File: static/app/views/issueDetails/streamline/foldSection.tsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+  return (
+    <section {...props}>
+      <Details open={!isCollapsed || preventCollapse}>
+        <Summary
+          preventCollapse={preventCollapse}
+          onClick={preventCollapse ? e => e.preventDefault() : toggleCollapse}
+        >
+          <div>{title}</div>
+          <IconWrapper preventCollapse={preventCollapse}>
```
Comment (malwilley) [inline static/app/views/issueDetails/streamline/foldSection.tsx:54]: There should be a button somewhere with an appropriate aria label, as-is there aren't any clickable elements

File: static/app/components/globalDrawer/index.stories.tsx
Comment (leeandher) [reply static/app/components/globalDrawer/index.stories.tsx:1]: Yeah, referenced it when putting together the other story book and corrected a typo.

File: static/app/views/issueDetails/streamline/foldSection.tsx (hunk 2 above)
Comment (leeandher) [reply static/app/views/issueDetails/streamline/foldSection.tsx:54]: `<summary>` is a conventional clickable HTML element that [most often gets implicit roles](https://www.w3.org/TR/html-aria/#docconformance), but there was a note on it:

image

That said, there's also [an example](https://www.w3.org/TR/html-aria/#side-effects) implying not to change set the role to button, so I guess no action needed here? 

image

Though I might be misunderstanding.

File: static/app/views/issueDetails/streamline/foldSection.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+          sectionKey={FoldSectionKey.HIGHLIGHTS}
+        >
+          <Lorem />
+        </FoldSection>
+        <FoldSection
+          title={
+            <Title>
+              <div>Header with Actions</div>
+              <ButtonBar gap={1}>
+                <Button size="xs" icon={<IconAdd />}>
```
Comment (roggenkemper) [inline static/app/views/issueDetails/streamline/foldSection.stories.tsx:79]: might be because it's a test button, but clicking these buttons collapses the section

File: static/app/views/issueDetails/streamline/foldSection.tsx [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+                onClick={e => e.stopPropagation()}
+                onMouseEnter={() => setIsLayerEnabled(false)}
+                onMouseLeave={() => setIsLayerEnabled(true)}
+              >
+                {actions}
+              </div>
+            )}
+          </TitleWithActions>
+          <IconWrapper preventCollapse={preventCollapse}>
+            <IconChevron direction={isCollapsed ? 'down' : 'up'} size="xs" />
```
Comment (roggenkemper) [inline static/app/views/issueDetails/streamline/foldSection.tsx:129]: the chevrons on the stories page are pointing in the wrong direction

Comment (leeandher) [reply static/app/views/issueDetails/streamline/foldSection.tsx:129]: Good catch!  Updated the storybook video, but it should be accurate now.

File: static/app/views/issueDetails/streamline/foldSection.tsx (hunk 2 above)
Comment (malwilley) [reply static/app/views/issueDetails/streamline/foldSection.tsx:128]: Ah no you're right, I'm just unfamiliar with the summary element!

File: static/app/views/issueDetails/streamline/foldSection.tsx [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+  sectionKey,
+  initialCollapse = false,
+  preventCollapse = false,
+  ...props
+}: FoldSectionProps) {
+  const [isCollapsed, setIsCollapsed] = useLocalStorageState(
+    `${LOCAL_STORAGE_PREFIX}${sectionKey}`,
+    initialCollapse
+  );
+  const [isLayerEnabled, setIsLayerEnabled] = useState(true);
```
Comment (malwilley) [inline static/app/views/issueDetails/streamline/foldSection.tsx:99]: Why is `isLayerEnabled` necessary?

File: static/app/views/issueDetails/streamline/foldSection.stories.tsx
Comment (malwilley) [inline static/app/views/issueDetails/streamline/foldSection.stories.tsx:1]: Very nice 👏

File: static/app/views/issueDetails/streamline/foldSection.tsx (hunk 5 above)
Comment (leeandher) [reply static/app/views/issueDetails/streamline/foldSection.tsx:99]: I wasn't sure of another way to accomplish this, but it's used to disable the interaction state layer when hovering over the 'button area', to keep things legible.

https://github.com/user-attachments/assets/7f61aabc-94d8-4806-836a-81dcb5296d15

Comment (leeandher) [reply static/app/views/issueDetails/streamline/foldSection.tsx:99]: Is this still the case, i wasn't able to repro

https://github.com/user-attachments/assets/4aed6e83-1e77-4773-a3d6-4a29e265b9e4

Comment (malwilley) [reply static/app/views/issueDetails/streamline/foldSection.tsx:99]: Ahh I see! If you can't find another way around that I'd add a comment.

One other thing you could try would be styling the header with something like:

```css
&:has(button:hover) {
  ${InteractionStateLayer} {
    display: none;
  }
}
```

Not sure if that actually works or not (or is any better)

File: static/app/views/issueDetails/streamline/foldSection.stories.tsx (hunk 3 above)
Comment (leeandher) [reply static/app/views/issueDetails/streamline/foldSection.stories.tsx]: Is this still the case, i wasn't able to repro

https://github.com/user-attachments/assets/4aed6e83-1e77-4773-a3d6-4a29e265b9e4

**Review by malwilley — APPROVED** [review]

File: static/app/views/issueDetails/streamline/foldSection.stories.tsx (hunk 3 above)
Comment (roggenkemper) [reply static/app/views/issueDetails/streamline/foldSection.stories.tsx]: nope - it looks good now!

**Review by roggenkemper — APPROVED** [review]

File: static/app/views/issueDetails/streamline/foldSection.tsx [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+
+  const toggleCollapse = useCallback(
+    (e: React.MouseEvent) => {
+      e.preventDefault(); // Prevent browser summary/details behaviour
+      setIsCollapsed(collapsed => {
+        trackAnalytics('issue_details.section_fold', {
```
Comment (malwilley) [inline static/app/views/issueDetails/streamline/foldSection.tsx:105]: I don't know if there is a rule for this or not, but I'd put the `trackAnalytics` call outside of the state updater function. It should be idempotent
